### PR TITLE
Return nullptr if no ServerConnection or any ClientConnections exist

### PR DESCRIPTION
**Pull request by Vatyx**
#### Description
We now return nullptr if no NetConnections exist at all. Added some other null checks elsewhere which rely on this.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 1]
```diff
@@ -419,6 +419,12 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 			}
 		}
 
+		if (Connection == nullptr)
+		{
+			UE_LOG(LogSpatialReceiver, Error, TEXT("Unable to find SpatialOSNetConnection! Have you timed out?"));
```
Comment (padhansell) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: "Has this worker been disconnected from SpatialOS due a timeout?"?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 2]
```diff
@@ -1112,7 +1112,7 @@ void USpatialNetDriver::ProcessRemoteFunction(
 		return;
 	}
 
-	USpatialNetConnection* NetConnection = ServerConnection ? Cast<USpatialNetConnection>(ServerConnection) : GetSpatialOSNetConnection();
+	USpatialNetConnection* NetConnection = GetSpatialOSNetConnection();
 	if (NetConnection == nullptr)
 	{
 		UE_LOG(LogSpatialOSNetDriver, Error, TEXT("Attempted to call ProcessRemoteFunction before connection was established"));
```
Comment (padhansell) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: "Attempted to call ProcessRemoteFunction but no SpatialOSNetConnection existed"?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 1 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Changed

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 2 above)
Comment (Vatyx) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Changed

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 3]
```diff
@@ -1540,6 +1544,11 @@ USpatialActorChannel* USpatialNetDriver::CreateSpatialActorChannel(AActor* Actor
 {
 	USpatialActorChannel* Channel = nullptr;
 
+	if (InConnection == nullptr)
+	{
+		return Channel;
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: bullptrpls

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 4]
```diff
@@ -1538,6 +1542,11 @@ USpatialActorChannel* USpatialNetDriver::GetActorChannelByEntityId(Worker_Entity
 
 USpatialActorChannel* USpatialNetDriver::CreateSpatialActorChannel(AActor* Actor, USpatialNetConnection* InConnection)
 {
+	if (InConnection == nullptr)
+	{
+		return nullptr;
```
Comment (joshuahuburn) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: bullptr

**Comment by oblm** [comment]
This corresponds to UNR-1566

**Pull request merged**